.github/workflows/unit_tests.yml

```diff
@@ -169,6 +169,12 @@ jobs:
           PRINTED_MSG=$(egrep -v "${IGNORE_PATTERNS}" test_framework_suite.log | grep '\.\n*[A-Za-z]' || true)
           test "x$PRINTED_MSG" = "x" || (echo "ERROR: Found printed messages in output of test suite\n${PRINTED_MSG}" && exit 1)
 
+          # There may be encoding errors in Python 3 which are hidden when an UTF-8 encoding is set
+          # Hence run the tests again with LC_ALL=C
+          if [[ ${{matrix.python}} =~ "3." ]]; then
+            LC_ALL=C python -O -m test.framework.suite
```

Comment:
Oh may, there's a bit extreme... The test suite already runs quite long.

How about just adding a test configuration with Python 3.x + `$LC_ALL` set to `C`?

Reply:
Yeah, it's another 16 minutes on top of the 18 minutes...
IMO we need this for Python 3.5 and any Python >= 3.6. At least till we find another issue... Hence this is the most complete and the safe bet.
But running tests with LC_ALL=C is also potentially problematic. E.g. I found issues where the tests relied on an English locale which gets hidden by LC_ALL=C.
But yes, I guess it makes sense to refine the test matrix. But then we should talk about it in more general. E.g.

> exclude Python 3.x versions other than 3.6, to limit test configurations

This is not what's happening now and the `exclude` is overly complicated. `include` is much better here. So maybe we should first define which test configurations to test exactly. This should include Python 3.5 and any Python >= 3.6 run twice for e.g. LC_ALL and a non-English UTF-8 locale

Reply:
We currently have 3 "dimensions":

* Python version (currently 6)
* modules tool (5)
* module syntax (2)

So in theory we have 60 combinations, but some don't make sense (non-Lmod modules with Lua syntax), and we already strip out some to limit the number of test configurations (we only test Lmod 8 with all Python 3 versions), which leads to 22 test configurations currently (+ 1 code style check).

Adding environment variables like `$LC_ALL` in the mix blows things up, especially if want want to consider multiple values for `$LC_ALL` (default, `C`, non-English UTF-8).

I would be fine, for now, with just adding `LC_ALL=C` with Python 3.6, Lmod 8 and Lua, since that would be enough to catch the problem being fixed, and it's the most common setup in which EasyBuild is being used (the exact Python 3.x version doesn't matter much).

Doing that would indeed require a rewrite of the test matrix, which I'd like to keep for a separate PR (we really want this one merged for the next EasyBuild release, which I want to push out ASAP now).

So how about we make the `if` test here a stricter so that we only also check with `LC_ALL=C` in *one* test configuration? Then that configuration would be "expensive", but it's only a single one...

Ideally we also change the order of the values in the test matrix being tested, so the long-running test configuration is picked up first . Not all test configurations start in parallel, so the expensive one(s) should start first so they can overlap with others and not block the outcome of the whole test suite.

Reply:
Another option would be to just add a specific test for using `read_file` & co when `LC_ALL=C` is set, but that may not be entirely trivial (you probably have to run the tests in a Python subshell to actually reproduce the problem being fixed here).

Reply:
And the issue is in more than just read_file. So we need to run all of it to be sure.

Reply:
I added an extra job for that